pkg/store/github/github.go

```diff
@@ -67,6 +67,7 @@ type GitHubStore struct {
 	moduleCache           map[string][]*core.ModuleVersion
 	providerVersionsCache map[string]*core.ProviderVersions
 	providerCache         map[string]*core.Provider
+	providerIgnoreCache   map[string]bool
```

Comment:
Here we should use a `sync.Map`. This is an append-only cache and synchronisation is, not critical, but necessary and cheap.